data/elan-2514-alt2.tablet

```diff
@@ -0,0 +1,19 @@
+# i2c:04f3:25bf
+# HP Spectre x360 Convertible 15t-ch000
+#
+# https://github.com/linuxwacom/wacom-hid-descriptors/issues/128
+
+[Device]
+Name=ELAN 2514
+ModelName=
+DeviceMatch=i2c:04f3:25bf;
+Class=ISDV4
+Width=14
+Height=8
+IntegratedIn=Display;System
+Styli=0xffffd;0xffffe;0xfffff;0x21;0x31;0x49;0x71;0x271;0x221;0x421;0x806B;0x807B;0x826B;0x846B;
```

Comment:
This looks like the AES list, isn't it? If so, just use `@isdv-aes` here and this way we're future-proof for new AES pens too.

Reply:
well, the screen I think is supposed to be MPP rather than AES, however I know a lot of the AES pens listed work for this laptop (such as the HP Active pens and the Wacom Bamboo Ink/Ink Plus). I just updated it to that tho just in case. Least with the pen I'm currently using tho, even with that it's still not showing in the Gnome Stylus settings (the digi does appear tho), so I'm not sure what's going on there.
Thanks!

Reply:
fwiw, libwacom has no effect on the *functionality* of the tablet itself, merely on whether it shows up in the control center and whether gnome maps it to the correct screens, etc. Pressure is not affected by libwacom, but we can debug this separately in libinput or xf86-input-wacom, depending on whether you use X or wayland.

I'd say let's go with the aes list (actual technology doesn't matter, it's just the name we picked to group those pens) until we find it's incorrect and let's fix it then.

However, if the stylus doesn't appear then that is a problem we should address. libinput has the `sudo ./build/libinput-debug-tablet` tool (might be installed by your distribution) that shows the stylus id while in proximity. does that show the correct values?

Reply:
> fwiw, libwacom has no effect on the _functionality_ of the tablet itself, merely on whether it shows up in the control center and whether gnome maps it to the correct screens, etc. Pressure is not affected by libwacom, but we can debug this separately in libinput or xf86-input-wacom, depending on whether you use X or wayland.
> 
> I'd say let's go with the aes list (actual technology doesn't matter, it's just the name we picked to group those pens) until we find it's incorrect and let's fix it then.
> 
> However, if the stylus doesn't appear then that is a problem we should address. libinput has the `sudo ./build/libinput-debug-tablet` tool (might be installed by your distribution) that shows the stylus id while in proximity. does that show the correct values?
ahh, okay. I just changed the Styli string to `@isdv4-aes` and it should have updated it here. I am on X, the exact distro is Pop_OS 20.04.

I just got me a Wacom Bamboo Ink Plus (thinking that my Raphael 520 might have been too generic), and both show this in `libinput debug-tablet`:
```
Device: ELAN2514:00 04F3:25BF (event12)                                         
Tool: pen serial 0, id 0                                             
libinput:
tip: up                                                                         
  x:                  33.63 [-----|-------------------------------------------]
  y:                  35.21 [---------|---------------------------------------]
  tilt x:              0.00 [------------------------|------------------------]
  tilt y:              0.00 [------------------------|------------------------]
  dist:                0.00 [|------------------------------------------------]
  pressure:            0.00 [|------------------------------------------------]
  rotation:            0.00 [|------------------------------------------------]
  slider:              0.00 [------------------------|------------------------]
  buttons:                                                                      
evdev:
  ABS_X:            2077.00 [-----|-------------------------------------------]
  ABS_Y:            2176.00 [---------|---------------------------------------]
  ABS_Z:               0.00 [|------------------------------------------------]
  ABS_TILT_X:          0.00 [|------------------------------------------------]
  ABS_TILT_Y:          0.00 [|------------------------------------------------]
  ABS_DISTANCE:        0.00 [|------------------------------------------------]
  ABS_PRESSURE:        0.00 [|------------------------------------------------]
  buttons: BTN_TOOL_PEN
```
The Bamboo is also not wanting to pair (it connects for half a second then disconnects, but I'm not sure if that's related to this) nor is it showing in the Wacom Stylus settings (and I did update my libwacom to the updated styli setting).

Reply:
note that this is an interactive tool (as you probably noticed). For the stylus id identification, the `Tool:` line is the one that matters, it'll update whenever a tool is in proximity. Does it show a serial/pen ID when in proximity? Looks like at least the other values seem correct though.

Reply:
Nope, with both the Raphael and the Wacom it just lists `Tool: pen serial 0, id 0` when they're in proximity.

Reply:
Then I'd guess either the device doesn't support specific styli or your stylus doesn't have an ID. You can remove the `Stylus=` line, IIRC that's the best option here. The other ELAN devices we already have, the majority of them don't have a specified stylus list either.

Reply:
FTR, this conversation sparked #340, not ready yet but you can see what it'll be useful for :)